src/foam/box/SessionServerBox.java

```diff
@@ -85,12 +85,23 @@ public void send(Message msg) {
           return;
         }
 
+        // If there is no user in the session (which happens when a user is
+        // signing up, for example) then set the URL in the app configuration to
+        // the URL that the request is coming from.
+        if ( req != null && ! SafetyUtil.isEmpty(req.getRequestURI()) ) {
+          AppConfig appConfig = (AppConfig) x.get("appConfig");
+          appConfig = (AppConfig) appConfig.fclone();
+          String configUrl = ((Request) req).getRootURL().toString();
```

Comment:
So the url at this point http, while in front of the load balancers, for example it's https.  There is a flag in appConfig forceHttps which you need to use to craft the url correctly.